api/controllers/login.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const https = require('https')
+const jwt = require('jsonwebtoken')
+const logger = require('../../config/logger')
+const util = require('../util/util')
+
+/* istanbul ignore next */
+const get = (request, response) => {
+  request.body = {
+    ticket: request.query['ticket']
+  }
+
+  post(request, response)
+}
+
+const post = (request, response) => {
+  var serviceTicket = request.body['ticket']
+
+  if (!serviceTicket) {
+    util.buildUnauthorizedResponse(response)
+    return
+  }
+
+  validateTicket(serviceTicket, (data) => {
+    if (!data) {
+      util.buildInternalErrorResponse(response)
+      return
+    }
+
+    var json = JSON.parse(data)
+    if (json.serviceResponse.authenticationFailure) {
+      logger.debug('Login failure: ' + json.serviceResponse.authenticationFailure.description)
+      util.buildUnauthorizedResponse(response)
+      return
+    }
+    var guid = json.serviceResponse.authenticationSuccess.attributes.theKeyGuid[0].toLowerCase()
```

Comment:
I would use `ssoGuid` instead of `theKeyGuid`.

They will be the same value, and `theKeyGuid` is present to facilitate migrations from `relayGuid` to `theKeyGuid`. At some point we might deprecate `theKeyGuid` once everything is migrated.

Reply:
Will `ssoGuid` ever be the Relay guid when the two are different? Or will it always be The Key guid?

Reply:
it will only ever be `ssoGuid` for an individual service if we configure that individual service to release the relayGuid as the `ssoGuid`. So far we have never used that configuration, and I'm hoping that services will do the migration work before they switch so we don't have to actually ever use that configuration option.

Reply:
How the content scoring API is configured :)
<img width="910" alt="screen shot 2018-03-27 at 4 58 56 pm" src="https://user-images.githubusercontent.com/47818/37994644-2cbf89d4-31e0-11e8-8b5e-308c9e084728.PNG">